integration-tests/src/test/java/oracle/weblogic/kubernetes/ItHorizontalPodAutoscalerCustomMetrics.java

```diff
@@ -0,0 +1,391 @@
+    installMonitoringExporter(monitoringExporterDir);
+    logger.info("create and verify WebLogic domain image using model in image with model files");
+    String miiImage = MonitoringUtils.createAndVerifyMiiImage(monitoringExporterAppDir,
+        MODEL_DIR + "/" + MONEXP_MODEL_FILE,
+        SESSMIGR_APP_NAME, MONEXP_IMAGE_NAME);
+    HashMap<String, String> labels = new HashMap<>();
+    labels.put("app", "monitoring");
+    labels.put("weblogic.domainUid", "test");
+    String className = "ItMonitoringExporterWebApp";
+
+    logger.info("create pv and pvc for monitoring");
+    assertDoesNotThrow(() -> createPvAndPvc(prometheusReleaseName, monitoringNS, labels, className));
+    assertDoesNotThrow(() -> createPvAndPvc("alertmanager" + releaseSuffix, monitoringNS, labels, className));
+    cleanupPromGrafanaClusterRoles(prometheusReleaseName, null);
+    cleanupPrometheusAdapterClusterRoles();
+
+    domain = createDomainResource(
+        domainUid,
+        domainNamespace,
+        miiImage,
+        adminSecretName,
+        new String[]{TEST_IMAGES_REPO_SECRET_NAME},
+        encryptionSecretName
+    );
+
+    // install and verify NGINX
+    nginxHelmParams = installAndVerifyNginx(nginxNamespace, 0, 0);
+
+    String nginxServiceName = nginxHelmParams.getHelmParams().getReleaseName() + "-ingress-nginx-controller";
+    logger.info("NGINX service name: {0}", nginxServiceName);
+    nodeportshttp = getServiceNodePort(nginxNamespace, nginxServiceName, "http");
+    logger.info("NGINX http node port: {0}", nodeportshttp);
+
+    // create cluster resouce with limits and requests in serverPod
+    ClusterResource clusterResource =
+        createClusterResource(clusterResName, wlClusterName, domainNamespace, replicaCount);
+    clusterResource.getSpec()
+        .serverPod(new ServerPod().resources(
+            new V1ResourceRequirements()
+                .putLimitsItem("cpu", Quantity.fromString("2"))
+                .putLimitsItem("memory", Quantity.fromString("2Gi"))
+                .putRequestsItem("cpu", Quantity.fromString("250m"))
+                .putRequestsItem("memory", Quantity.fromString("768Mi"))));
+    logger.info("Creating cluster {0} in namespace {1}", clusterResName, domainNamespace);
+    createClusterAndVerify(clusterResource);
+
+    // set cluster references
+    domain.getSpec().withCluster(new V1LocalObjectReference().name(clusterResName));
+
+    // verify the domain custom resource is created
+    createDomainAndVerify(domain, domainNamespace);
+
+    // check admin server is up and running for domain1
+    checkPodReadyAndServiceExists(adminServerPodName, domainUid, domainNamespace);
+
+    // check managed servers are up and running for domain1
+    for (int i = 1; i <= replicaCount; i++) {
+      checkPodReadyAndServiceExists(managedServerPrefix + i, domainUid, domainNamespace);
+    }
+  }
+
+
+  /**
+   * Test autoscaling using HPA by increasing the CPU usage.
```

Comment:
Need more description with respect to usage of Monitoring Exporter

Reply:
done